### PR TITLE
Move swagger codegen to separate folder.

### PR DESCRIPTION
**Pull request by weeksling**
feat: Separate strava API into separate package and make next project main.

 - Create strave_api_v3 to match strava docs
 - Add start to yarn workspaces config

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-  "dependencies": {
-    "@types/node": "8.10.34",
-    "bluebird": "^3.7.2",
-    "http": "0.0.1-security"
-  },
-  "devDependencies": {
-    "ts-node": "^8.6.2",
-    "typescript": "^3.8.3"
-  }
+  "workspaces": [
```
Comment (Deborah-Digges) [inline package.json:11]: nice!

File: strava_api_v3/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "@types/node": "^8",
+    "bluebird": "^3.7.2",
+    "http": "0.0.1-security",
+    "request": "^2.88.2"
+  },
+  "devDependencies": {
+    "ts-node": "^8.10.2",
```
Comment (Deborah-Digges) [inline strava_api_v3/package.json:21]: I was using ts-node initially to test things out.

I wonder if we'd need to remove this eventually when we extract this as a publishable package

File: strava_api_v3/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+var api_1 = require("./api");
+var StravaAPI = require("./api");
```
Comment (Deborah-Digges) [inline strava_api_v3/lib/index.js:4]: Curious about this bit. Why do we need the extra `api_1` variable?

File: strava_api_v3/src/index.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
 import { ActivitiesApi, OAuth, Authentication, AthletesApi } from "./api";
+import * as StravaAPI from "./api"
```
Comment (Deborah-Digges) [inline strava_api_v3/src/index.ts:2]: `index.ts → strava_api_v3/src/index.ts` and `strava_api_v3/lib/index.js` seem to be awfully similar. Do we need em both?

Comment (weeksling) [reply strava_api_v3/src/index.ts:2]: Nope, lib should be ignored 😅 That's just the `yarn build` output

File: strava_api_v3/lib/index.js (hunk 3 above)
Comment (weeksling) [reply strava_api_v3/lib/index.js:4]: This is generated by typescript! I'll remove it actually and .gitignore lib

File: strava_api_v3/package.json (hunk 2 above)
Comment (weeksling) [reply strava_api_v3/package.json:21]: I think as a dev dependency it's fine. To run it in scripts regardless we need tsnode (preferably used via `--register`) and tsc to compile. For actually publishing we might need to do some magic-ness around compiling and publishing the build only. I added a task to our checklist for making it publishable.

File: package.json (hunk 1 above)
Comment (weeksling) [reply package.json:11]: Who knows if it works! 😅

File: strava_api_v3/lib/index.js (hunk 3 above)
Comment (Deborah-Digges) [reply strava_api_v3/lib/index.js:4]: ohh I didn't realize it was generated :octocat:

Comment (lazerblasters) [reply strava_api_v3/lib/index.js]: Thanks for catching that!

**Pull request merged**